Dockerfile

```diff
@@ -11,6 +11,16 @@ FROM python:3.7-slim-buster AS builder
 SHELL ["/bin/bash", "-c"]
 
 # Add system deps for building
+# autoconf, automake: for building VCFtools; may be used by package managers to build from source
+# build-essential: contains gcc, g++, make, etc. for building various tools; may be used by package managers to build from source
+# ca-certificates: ?
+# curl: for downloading source files
+# git: for git pip installs
+# jq: used in devel/latest-augur-release-tag
+# libgmp-dev: ?
+# libpng-dev: ?
+# pkg-config: ?
```

Comment:
@tsibley do you remember why the packages with `?` were added in 3917c1b9aa0aa19e694f94f6bd3e56418f248269?

Reply:
These would have been for compiling deps. Not sure which ones, possibly ones a) that have since been removed or b) which we no longer compile ourselves (e.g. cvxopt, but there are also others).

Note that the runtime library versions of these packages (i.e. the non `-dev` versions, e.g. `libgmp10` and `libpng`) were removed in the switch from Alpine → Debian in ce07dadb5f6e2786d6ad9eabef00745b501ed425, which implies (b) and static linking.

One way to find out is to checkout 3917c1b and (after verifying you can build the image as-it-was), try building it without those deps and see where it fails.

IIRC, both Augur and Auspice had transitive dependencies on libpng at one point, though this may have since been loosened.

Generally speaking, GMP is often required for bignum support in numerical software, so it's not surprising to see. (I'm a little surprised we don't need it at runtime, but static linking could explain this. Slightly terrifyingly, I've run into libraries that make their use of a bignum library like GMP optional, falling back to a horrifyingly slow implementation when GMP isn't available. Hopefully that's not the case here!)

Reply:
I've tested these on the current Dockerfile:

- `pkg-config` is needed by VCFtools. Noted in a9dd473.
- `libgmp-dev` is not needed anymore. Removed in 1bcd494.
- `libpng-dev` is not needed, but its dependency `zlib1g-dev` is needed by VCFtools. Updated in a47e358.

I checked out 3917c1b9aa0aa19e694f94f6bd3e56418f248269, but building from that commit no longer works due to changes in upstream URL availability. I did not dig deeper to find out exactly where `libgmp-dev` and `libpng-dev` were used.

Reply:
> I've run into libraries that make their use of a bignum library like GMP optional, falling back to a horrifyingly slow implementation when GMP isn't available. Hopefully that's not the case here!

That doesn't sound good! Maybe worth adding back in a separate PR and doing trial runs on the ncov workflow?

Reply:
Nice testing work.

I definitely think it's a good idea to confirm that GMP's presence/absence in the builder and/or final image doesn't affect execution times. ncov is probably a good test bench? or maybe monkeypox?